framework/decode/vulkan_replay_consumer_base.cpp

```diff
@@ -969,8 +970,19 @@ void VulkanReplayConsumerBase::ProcessSetSwapchainImageStateQueueSubmit(
     pool_create_info.flags                   = VK_COMMAND_POOL_CREATE_RESET_COMMAND_BUFFER_BIT;
     pool_create_info.queueFamilyIndex        = queue_family_index;
 
-    // TODO: Improved queue selection?
-    table->GetDeviceQueue(device, queue_family_index, 0, &queue);
+    const auto queue_family_flags = device_info->queue_family_creation_flags.find(queue_family_index);
+    assert(queue_family_flags != device_info->queue_family_creation_flags.end());
+    if (queue_family_flags->second)
```

Comment:
Since this is a bitmask, can this be more explicit, like `if (queue_family_flags->second != 0)`?

Reply:
Sure